src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -500,17 +500,21 @@ public static Material getBlockMaterial(ResourceLocation loc)
      * internal, relies on fixed format of FaceBakery
      */
     // TODO Do we need this?
-    public static void fillNormal(int[] faceData, Direction facing)
-    {
-        Vector3f v1 = getVertexPos(faceData, 3);
-        Vector3f t1 = getVertexPos(faceData, 1);
-        Vector3f v2 = getVertexPos(faceData, 2);
-        Vector3f t2 = getVertexPos(faceData, 0);
-        v1.sub(t1);
-        v2.sub(t2);
-        v2.cross(v1);
-        v2.normalize();
-
+    public static void fillNormal(int[] faceData, Direction facing, boolean calculateNormals)
+    {
+        Vector3f v2;
+        if (calculateNormals) {
+            Vector3f v1 = getVertexPos(faceData, 3);
+            Vector3f t1 = getVertexPos(faceData, 1);
+            v2 = getVertexPos(faceData, 2);
+            Vector3f t2 = getVertexPos(faceData, 0);
+            v1.sub(t1);
+            v2.sub(t2);
+            v2.cross(v1);
+            v2.normalize();
+        } else {
+            v2 = new Vector3f(facing.getNormal().getX(), facing.getNormal().getY(), facing.getNormal().getZ());
+        }
```

Comment:
bincompat: Include the old method with the previous signature `(int[], Direction)` and call the new method with a default argument (example: `false`).